soccer/Processor.cpp

```diff
@@ -510,119 +475,27 @@ void Processor::run() {
-
-    // rotation
-    setpoint.avelocity = controlVals.rotation;
-
-    // kick/chip
-    bool kick = controlVals.kick || controlVals.chip;
-    intent.trigger_mode =
-        (kick ? (_context.game_settings.joystick_config.useKickOnBreakBeam
-                     ? RobotIntent::TriggerMode::ON_BREAK_BEAM
-                     : RobotIntent::TriggerMode::IMMEDIATE)
-              : RobotIntent::TriggerMode::STAND_DOWN);
-    intent.kcstrength = static_cast<int>(controlVals.kickPower);
-    intent.shoot_mode = (controlVals.kick ? RobotIntent::ShootMode::KICK
-                                          : RobotIntent::ShootMode::CHIP);
-
-    // dribbler
-    intent.dvelocity =
-        static_cast<float>(controlVals.dribble ? controlVals.dribblerPower : 0);
-}
-
-JoystickControlValues Processor::getJoystickControlValue(Joystick& joy) {
-    JoystickControlValues vals = joy.getJoystickControlValues();
-    if (joy.valid()) {
-        // keep it in range
-        vals.translation.clamp(sqrt(2.0));
-        if (vals.rotation > 1) {
-            vals.rotation = 1;
-        }
-        if (vals.rotation < -1) {
-            vals.rotation = -1;
-        }
-
-        // Gets values from the configured joystick control
-        // values,respecting damped
-        // state
-        if (_context.game_settings.joystick_config.dampedTranslation) {
-            vals.translation *=
-                Joystick::JoystickTranslationMaxDampedSpeed->value();
-        } else {
-            vals.translation *= Joystick::JoystickTranslationMaxSpeed->value();
-        }
-        if (_context.game_settings.joystick_config.dampedRotation) {
-            vals.rotation *= Joystick::JoystickRotationMaxDampedSpeed->value();
-        } else {
-            vals.rotation *= Joystick::JoystickRotationMaxSpeed->value();
-        }
-
-        // scale up kicker and dribbler speeds
-        vals.dribblerPower *= Max_Dribble;
-        vals.kickPower *= Max_Kick;
+        MotionSetpoint& setpoint = _context.motion_setpoints[r->shell()];
+
+        setpoint.clear();
+        intent.dvelocity = 0;
+        intent.kcstrength = 0;
+        intent.shoot_mode = RobotIntent::ShootMode::KICK;
+        intent.trigger_mode = RobotIntent::TriggerMode::STAND_DOWN;
+        intent.song = RobotIntent::Song::STOP;
     }
-    return vals;
 }
 
-std::vector<JoystickControlValues> Processor::getJoystickControlValues() {
-    std::vector<JoystickControlValues> vals;
-    for (Joystick* joy : _joysticks) {
-        if (joy->valid()) {
-            vals.push_back(getJoystickControlValue(*joy));
-        }
+void Processor::updateIntentActive() {
+    // Intent is active if it's being joystick controlled, or
+    // if it's visible.
+    for (OurRobot* r : _context.state.self) {
+        RobotIntent& intent = _context.robot_intents[r->shell()];
+        intent.is_active = r->isJoystickControlled() || r->visible();
     }
-    return vals;
 }
 
 void Processor::recalculateWorldToTeamTransform() {
```

Comment:
I just feel like I need to do a :tada: every time we cut code from Processor.

That's all.